proxy/common/types.py

```diff
@@ -10,10 +10,17 @@
 """
 import queue
 import ipaddress
+import sys
 
 from typing import TYPE_CHECKING, Dict, Any, List, Union
 
-from typing_extensions import Protocol
+# NOTE: Using try/except causes linting problems which is why it's necessary
+# NOTE: to use this mypy/pylint idiom for py36-py38 compatibility
+# Ref: https://github.com/python/typeshed/issues/3500#issuecomment-560958608
+if sys.version_info >= (3, 8):
+    from typing import Protocol
+else:
+    from typing_extensions import Protocol
```

Comment:
Thank you :). Was totally unnecessary to include typings for all future Python versions